tests/Unit/DataStructures/Test_DataVector.cpp

```diff
@@ -53,7 +53,9 @@ SPECTRE_TEST_CASE("Unit.Serialization.DataVector",
   DataVector t(npts), tgood(npts);
   std::iota(t.begin(), t.end(), 1.2);
   std::iota(tgood.begin(), tgood.end(), 1.2);
-  CHECK(tgood == serialize_and_deserialize(t));
```

Comment:
Looking at this test, it wasn't good enough before. The owning-ness should be tested after serialization, so I don't think the `test_serialization` function helps. We probably want:

```cpp
CHECK(tgood == t);
const DataVector serilaized = serialize_and_deserialize(t);
CHECK(tgood = t);
CHECK(serialized == tgood);
CHECK(serialized.owning());
CHECK(serialized.data() != t.data());
CHECK(/*owning status of t*/);
```